snuba/datasets/processors/transactions_processor.py

```diff
@@ -472,4 +474,11 @@ def process_message(
         # the following operation modifies the event_dict and is therefore *not* order-independent
         self._process_contexts_and_user(processed, event_dict)
 
-        return InsertBatch([processed], None)
+        try:
+            raw_received = _collapse_uint32(int(event_dict["data"]["received"]))
+            assert raw_received is not None
```

Comment:
If received is not there, won't you get an AssertionError that isn't being caught?

Reply:
Right, but wouldn't the first line (which throws KeyError) be thrown first?

Reply:
But I guess the AssertionError could be even easier, since it'd be guaranteed that it's being thrown at that line